x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/create/index.tsx

```diff
@@ -330,6 +333,7 @@ const CreateRulePageComponent: React.FC = () => {
                   setForm={setFormHook}
                   onSubmit={submitStepDefineRule}
                   descriptionColumns="singleSplit"
+                  key={isShouldRerenderStep(RuleStep.defineRule)}
```

Comment:
Could you explain what this does exactly? I'm not sure I understand the implication of setting key to `0` or `1` on all of these.

Reply:
Inside each step, we have some structure like:
```
const Step = ( {isReadOnlyView}) => {
     const form = useForm({defaultValues});
     
     return isReadOnlyView ? <DetailsView> : <Form form={form}> </Form>; 
}
```

Previously when we change isReadOnlyView the `form` was reinitialised with values from the previous state. [But after some changes in the form](https://github.com/elastic/kibana/pull/130544), it doesn't happen again. ([Aslo comment about the hook](https://github.com/elastic/kibana/pull/130825#pullrequestreview-951482981))

So we need to remount this component to be able to create a proper form.  And key change is helping to remount it.


Another way of doing it in the more proper and declarative way it's to split the Card Details and Form component, and move `useForm` to the form component. But it's will introduce much more code changes, which not ideal for FF

Reply:
Gotcha! In that case it would be great to have a tiny comment on that line of code explaining the `key` usage and then maybe add a `//TODO` as well. Wdyt?